### ISSUE TITLE
Add date_verified and date_updated to supported metadata time conversions and blog layout

### ISSUE THREAD

**Issue by MasonEgger** [squidfunk/mkdocs-material#5733]
### Context

Documentation is known to get stale. SEO rewards content with metadata that is newer, rather than older. However,  changing the original publish date can affect this and is not a great solution. Many large content producers have added metadata that specifies the date of publication, as well as the last time the content was updated, and last time the content was verified. While this isn't necessary for blogs, it greatly helps reader confidence in technical content such as tutorials and documentation. 

Currently, the blog plugin supports a published `date`  that is automatically added to the blog post when built. This date field in the metadata currently benefits from being automatically converted to the set date format specified in the configuration file. However, if any other dates are added to the metadata file they do not get converted. 

I'm proposing adding a `date_verified` and `date_updated` field to the metadata that also get the benefit of date formatting and are applied to the blog post, similar to the `date` metadata option if set.

### Description

Add a date format normalization for two new metadata fields `date_updated` and `date_verified` (naming convention up to  you) to allow for these fields to be set properly in the metadata and automatically show up in the blog if set.

### Related links

- [Example of my personal site with the Verified and Updated fields](https://mason.dev/tutorials/how-to-build-a-7-days-to-die-server-on-ubuntu/)
- [Link to blog post with my hack for formatted dates](https://raw.githubusercontent.com/MasonEgger/website/main/docs/tutorials/posts/001-7d2d.md)
- [Example of a DigitalOcean Tutorial exposing the updated metadata](https://www.digitalocean.com/community/tutorials/how-to-redirect-www-to-non-www-with-nginx-on-centos-7)

### Use Cases

An author of the content would add the `date_verified` and `date_updated` keys to their metadata specifying when the content was last verified for correctness and updated respectively. Mkdocs-material would then format the dates similarly to how it does for the `date` field [here](https://github.com/squidfunk/mkdocs-material-insiders/blob/master/material/plugins/blog/plugin.py#L174) and make that variable available for the templates. There could also be additions made to the plugin to allow for this information to be automatically added to the metadata sidebar as shown in the image below. 

As a user, I would see that the documentation, tutorial, or blog that I'm reading is actively being maintained and have more confidence in its viability before continuing with it. 

This change would greatly impact the quality of the documentation for all who chose to use it, not only from a user experience standpoint but also from an SEO vantage as newer content tends to be ranked higher.

### Visuals

![blog-metadata](https://github.com/squidfunk/mkdocs-material/assets/11214847/df5c8b6f-6997-4252-88f6-83947bb91bc1)

### Before submitting

- [X] I have read and followed the [change request guidelines](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/).
- [X] I have verified that [my idea is a change request and not a bug report](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#its-not-a-bug-its-a-feature).
- [X] I have ensured that, to the best of my knowledge, [my idea will benefit the entire community](https://squidfunk.github.io/mkdocs-material/contributing/requesting-a-change/#benefit-for-the-community).
- [X] I have included relevant links to [the documentation](https://squidfunk.github.io/mkdocs-material), related [issues](https://github.com/squidfunk/mkdocs-material/issues), and [discussions](https://github.com/squidfunk/mkdocs-material/discussions) to underline the need for my idea.

**Comment by squidfunk** [comment]
Thanks for suggesting! There's already an open change request for `date_updated` in #5028, and that's definitely coming, but I'm hearing about `date_verified` for the first time. We'll let this request sit for a while to check if other users also think it would be useful 😊

**Comment by MasonEgger** [comment]
Awesome! Thanks! 

ya, `date_verified` isn't something people often think about and sometimes will just tie into date_updated. Imo though they are distinct because just because sometimes I will reuse my own tutorials to do something, find they are still up to date and relevant, and would like to let people know that just because it's a little old doesn't mean it doesn't work.

Hope this makes it in, but I have a workaround currently. Just wanted to mention incase others also were looking for a similar feature :)

**Comment by squidfunk** [comment]
Update: we're currently not considering adding `date_verified`, but we'll definitely add `date_updated`. As this was already requested in #5028, we can close this issue. Progress is tracked in #5208.

**Comment by squidfunk** [comment]
__Update__: we're rearchitected the blog plugin from the ground up, and it will be possible to add custom dates. As of 9.2 there will be two formats for front matter

### Single date (creation)

``` markdown
---
date: 2023-08-08
---

# Page
```

### Multiple dates

``` markdown
---
date:
  created: 2023-08-08 # <- must be set when using this syntax
  updated: 2023-08-09
  verified: 2023-08-10
  ...
---

# Page
```

Templates need to support adding arbitrary dates, and we can add this support in an upcoming version. Additionally, front matter is now validated using MkDocs regular configuration validation, which should make authoring much simpler.

**Comment by MasonEgger** [comment]
Fantastic! Thanks!

**Comment by squidfunk** [comment]
This is now part of the `merge/piri-piri` branch and will be released as part of 9.2.
